File: javascript/image-example.js
/**
 * x402 API Example: Image Generation with OpenAI GPT-Image-1
 * 
 * This example shows how to generate high-quality images using
 * OpenAI's GPT-Image-1 with automatic USDC payments.
 * 
 * Prerequisites:
 * 1. npm install
 * 2. node generate-wallet.js (get wallet address)
 * 3. Fund wallet with USDC on Base
 * 4. Create .env file with PRIVATE_KEY
 * 
 * Run: node javascript/image-example.js
 */

import { wrapFetchWithPayment } from 'x402-fetch';
import { privateKeyToAccount } from 'viem/accounts';
import 'dotenv/config';

// Configuration
const API_BASE_URL = process.env.API_BASE_URL || 'https://beatsx402.ai';
const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error('❌ Error: PRIVATE_KEY not found in .env file');
  console.log('\n📝 Setup:');
  console.log('1. Run: node generate-wallet.js');
  console.log('2. Copy .env.example to .env');
  console.log('3. Add your PRIVATE_KEY to .env\n');
  process.exit(1);
}

async function main() {
  try {
    console.log('🎨 x402 Image Generation Example\n');
    console.log('='.repeat(60));
    
    // 1. Set up wallet with x402
    const account = privateKeyToAccount(PRIVATE_KEY);
    const fetchWithPayment = wrapFetchWithPayment(fetch, account);
    
    console.log(`\n💰 Wallet: ${account.address}`);
    console.log('💸 Cost: $0.30 USDC');
    console.log('🔗 Network: Base');
    console.log('🤖 Model: OpenAI GPT-Image-1');
    
    // 2. Make request - x402 handles payment automatically
    console.log('\n📤 Generating image...\n');
    
    const response = await fetchWithPayment(`${API_BASE_URL}/v1/openai/images/generate`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        prompt: 'Cyberpunk koala DJ at neon-lit rooftop party overlooking futuristic city',
        size: '1024x1024',
        quality: 'high',
        background: 'opaque'
      })
    });
    
    const data = await response.json();
    
    // 3. Display response
    console.log('✅ Image generated!\n');
    console.log('='.repeat(60));
    
    if (data.images && data.images.length > 0) {
      console.log('🖼️  Image URL:');
      console.log(data.images[0].url);
      
      if (data.images[0].revised_prompt) {
        console.log('\n📝 Revised Prompt:');
        console.log(data.images[0].revised_prompt);
      }
    } else {
      console.log(JSON.stringify(data, null, 2));
    }
    
    console.log('='.repeat(60));
    
    console.log('\n💡 How x402 worked:');
    console.log('1. ✅ Initial request sent → Server: 402 Payment Required');
    console.log('2. ✅ x402-fetch signed $0.30 USDC payment on Base');
    console.log('3. ✅ Retried with X-PAYMENT header');
    console.log('4. ✅ Server verified payment & processed request');
    console.log('5. ✅ Payment settled in ~2 seconds\n');
    
  } catch (error) {
    console.error('❌ Error:', error.message);
    
    if (error.message.includes('402')) {
      console.log('\n💡 Payment issue - check wallet balance:');
      console.log('   - Ensure wallet has USDC on Base network');
      console.log('   - Minimum: $0.30 USDC for this request\n');
    }
  }
}

main();

File: python/image_example.py
"""
x402 API Example: Image Generation with OpenAI GPT-Image-1

This example shows how to generate high-quality images using
OpenAI's GPT-Image-1 with automatic USDC payments.

Prerequisites:
1. pip install -r requirements.txt
2. python generate-wallet.js (get wallet address)
3. Fund wallet with USDC on Base
4. Create .env file with PRIVATE_KEY

Run: python python/image_example.py
"""

from x402.clients.requests import x402_requests
from eth_account import Account
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Configuration
API_BASE_URL = os.getenv('API_BASE_URL', 'https://beatsx402.ai')
PRIVATE_KEY = os.getenv('PRIVATE_KEY')

if not PRIVATE_KEY:
    print('❌ Error: PRIVATE_KEY not found in .env file')
    print('\n📝 Setup:')
    print('1. Run: node generate-wallet.js')
    print('2. Copy .env.example to .env')
    print('3. Add your PRIVATE_KEY to .env\n')
    exit(1)


def main():
    try:
        print('🎨 x402 Image Generation Example\n')
        print('=' * 60)
        
        # 1. Set up wallet with x402
        account = Account.from_key(PRIVATE_KEY)
        session = x402_requests(account)
        
        print(f'\n💰 Wallet: {account.address}')
        print('💸 Cost: $0.30 USDC')
        print('🔗 Network: Base')
        print('🤖 Model: OpenAI GPT-Image-1')
        
        # 2. Make request - x402 handles payment automatically
        print('\n📤 Generating image...\n')
        
        response = session.post(
            f'{API_BASE_URL}/v1/openai/images/generate',
            json={
                'prompt': 'Cyberpunk koala DJ at neon-lit rooftop party overlooking futuristic city',
                'size': '1024x1024',
                'quality': 'high',
                'background': 'opaque'
            }
        )
        
        data = response.json()
        
        # 3. Display response
        print('✅ Image generated!\n')
        print('=' * 60)
        
        if 'images' in data and len(data['images']) > 0:
            print('🖼️  Image URL:')
            print(data['images'][0]['url'])
            
            if 'revised_prompt' in data['images'][0]:
                print('\n📝 Revised Prompt:')
                print(data['images'][0]['revised_prompt'])
        else:
            print(str(data))
        
        print('=' * 60)
        
        print('\n💡 How x402 worked:')
        print('1. ✅ Initial request sent → Server: 402 Payment Required')
        print('2. ✅ x402 client signed $0.30 USDC payment on Base')
        print('3. ✅ Retried with X-PAYMENT header')
        print('4. ✅ Server verified payment & processed request')
        print('5. ✅ Payment settled in ~2 seconds\n')
        
    except Exception as error:
        print(f'❌ Error: {str(error)}')
        
        if '402' in str(error):
            print('\n💡 Payment issue - check wallet balance:')
            print('   - Ensure wallet has USDC on Base network')
            print('   - Minimum: $0.30 USDC for this request\n')


if __name__ == '__main__':
    main()

